timelock-server/src/integTest/java/com/palantir/atlasdb/timelock/AsyncTimelockServiceIntegrationTest.java

```diff
@@ -41,111 +37,124 @@
-        LockTokenV2 token = CLUSTER.lock(requestFor(LOCK_A)).getToken();
-        Future<LockTokenV2> futureToken = CLUSTER.lockAsync(requestFor(LOCK_A))
+        LockTokenV2 token = cluster.lock(requestFor(LOCK_A)).getToken();
+        Future<LockTokenV2> futureToken = cluster.lockAsync(requestFor(LOCK_A))
                 .thenApply(LockResponseV2::getToken);
 
         assertNotYetLocked(futureToken);
 
-        CLUSTER.unlock(token);
+        cluster.unlock(token);
 
         assertLockedAndUnlock(futureToken);
     }
 
     @Test
     public void canLockImmutableTimestamp() {
-        LockImmutableTimestampResponse response1 = CLUSTER.timelockService()
+        LockImmutableTimestampResponse response1 = cluster.timelockService()
                 .lockImmutableTimestamp(LockImmutableTimestampRequest.create());
-        LockImmutableTimestampResponse response2 = CLUSTER.timelockService()
+        LockImmutableTimestampResponse response2 = cluster.timelockService()
                 .lockImmutableTimestamp(LockImmutableTimestampRequest.create());
 
-        long immutableTs = CLUSTER.timelockService().getImmutableTimestamp();
+        long immutableTs = cluster.timelockService().getImmutableTimestamp();
         assertThat(immutableTs).isEqualTo(response1.getImmutableTimestamp());
 
-        CLUSTER.unlock(response1.getLock());
+        cluster.unlock(response1.getLock());
 
         assertThat(immutableTs).isEqualTo(response2.getImmutableTimestamp());
 
-        CLUSTER.unlock(response2.getLock());
+        cluster.unlock(response2.getLock());
     }
 
     @Test
     public void immutableTimestampIsGreaterThanFreshTimestampWhenNotLocked() {
-        long freshTs = CLUSTER.getFreshTimestamp();
-        long immutableTs = CLUSTER.timelockService().getImmutableTimestamp();
+        long freshTs = cluster.getFreshTimestamp();
+        long immutableTs = cluster.timelockService().getImmutableTimestamp();
 
         assertThat(immutableTs).isGreaterThan(freshTs);
     }
 
     @Test
     public void canWaitForLocks() {
-        LockTokenV2 token = CLUSTER.lock(requestFor(LOCK_A, LOCK_B)).getToken();
+        LockTokenV2 token = cluster.lock(requestFor(LOCK_A, LOCK_B)).getToken();
 
-        CompletableFuture<WaitForLocksResponse> future = CLUSTER.waitForLocksAsync(waitRequestFor(LOCK_A, LOCK_B));
+        CompletableFuture<WaitForLocksResponse> future = cluster.waitForLocksAsync(waitRequestFor(LOCK_A, LOCK_B));
         assertNotDone(future);
 
-        CLUSTER.unlock(token);
+        cluster.unlock(token);
 
         assertDone(future);
         assertThat(future.join().wasSuccessful()).isTrue();
     }
 
     @Test
     public void canGetTimestamps() {
-        long ts1 = CLUSTER.getFreshTimestamp();
-        long ts2 = CLUSTER.getFreshTimestamp();
+        long ts1 = cluster.getFreshTimestamp();
+        long ts2 = cluster.timelockService().getFreshTimestamp();
 
         assertThat(ts2).isGreaterThan(ts1);
     }
```

Comment:
Nice, we're now also checking that whether you access timestamp directly or via timelock, the numbers are consistent.

Not sure if it makes sense to add an additional check where, after we get one from timelock, if we get one from regular timestamp that's even larger (the idea being that the changes propagate in both directions).